### PR TITLE
Implement pluggable serializers

### PR DESCRIPTION
**Pull request by dima-exe**
Hutch uses hardcoded JSON encoder/decoder for serialization. This patch adds pluggable serializers,  Protobuf/Thrift or something else may be used with Hutch.

### CODE REVIEW

**Comment by michaelklishin** [comment]
Can we do without this Rails-y DSL for managing serialisers? It's just a damn pair of functions, so lets assign them in the config and not waste CPU resources on every message published to look it up via a fancy DSL.

**Comment by dima-exe** [comment]
Done

File: lib/hutch/broker.rb [hunk 1]
```diff
@@ -358,5 +369,9 @@ def generate_id
     def global_properties
       Hutch.global_properties.respond_to?(:call) ? Hutch.global_properties.call : Hutch.global_properties
     end
+
+    def serializer(content_type)
```
Comment (teodor-pripoae) [inline lib/hutch/broker.rb]: Do you still need this method ?

**Comment by michaelklishin** [comment]
@dima-exe thank you. Please update [ChangeLog.md](https://github.com/gocardless/hutch/blob/master/CHANGELOG.md) and credit yourself!

**Comment by dkastner** [comment]
I just have to say it's awesome stuff like this that makes hutch such a great tool to work with. Thank you!

**Pull request merged**